File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/spi/TdengineTypeMappingProvider.java
package site.addzero.biz.spec.iot.spi;

import site.addzero.biz.spec.iot.IotPropertySpec;
import site.addzero.biz.spec.iot.IotValueType;
import site.addzero.biz.spec.iot.tdengine.TdColumnSpec;

/**
 * Maps generic telemetry value types to TDengine columns.
 */
public interface TdengineTypeMappingProvider {

    String getName();

    boolean supports(IotValueType valueType);

    TdColumnSpec toColumnSpec(IotPropertySpec propertySpec);
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/spi/TdengineTypeMappingProvider.kt
package site.addzero.biz.spec.iot.spi

import site.addzero.biz.spec.iot.IotPropertySpec
import site.addzero.biz.spec.iot.IotValueType
import site.addzero.biz.spec.iot.tdengine.TdColumnSpec

/**
 * Maps generic telemetry value types to TDengine columns.
 */
interface TdengineTypeMappingProvider {

    val name: String

    fun supports(valueType: IotValueType): Boolean

    fun toColumnSpec(propertySpec: IotPropertySpec): TdColumnSpec
}
